### DISCUSSION TITLE
ERROR: Missing 'psycopg' module from pip

### DISCUSSION THREAD

**Discussion by Rammurthy5** [pgEdge/cli#12] [Q&A] [answered]
Hey guys, 

I am trying to setup nodectl and spock in an ec2. I executed the following command and ended up with psycopg error (pip installed though). 
`./nodectl spock node-create n1 'host=10.1.2.5 user=pgedge dbname=demo' demo`
what am i doing wrong? thanks

**Comment by cmotyka** [comment]
Hey, just to get some background I wanted to verify-
You're EC2 is:
`EL9 (RHEL/CentOS/Rocky), AmazonLinux-2023, or Ubuntu-22.04`

and you installed with:
`python3 -c "$(curl -fsSL https://pgedge-download.s3.amazonaws.com/REPO/install.py)"`
`./nodectl install pgedge -U superuser-name -P superuser-password -d database-name`

**Reply by Rammurthy5** [reply]
yes, thats right. amazon linux 2023 image and followed the above two commands. i have tried for pg v15, and v16. both results in same error

**Comment by cmotyka** [comment]
Ok, we're going to be updating our getting started guide this week to reflect that we do not support Amazon Linux or Amazon Linux 2023. We're going to strongly suggest using Rocky Linux 9.2+, AMI image ids can be found here- https://rockylinux.org/cloud-images/

**Reply by Rammurthy5** [reply]
Sorry, was this tested at your end & found to be an issue? May I ask if this is an issue & be fixed in the future ?

**Reply by cmotyka** [reply]
Yes, there are other work arounds that our tech support organization can do, if you go through pgEdge.com

We are planning on supporting Amazon Linux 2023 later this year, is using Rocky Linux 9 a blocker to continuing to test?

**Reply by Rammurthy5** [reply]
Nope, I will try that out and get back if there are issues.

**Comment by luss** [comment]
Great stuff.  I'm gonna close this issue.  Note that I'me planning to add support for al2023 (at least for AMD) within the next few weeks.  It is after all EL9+ and it's primary difference from the EL9 standard is that it uses a newer version of python3 and a newer version of glibc (because it's Fedora based, on a version thats newer than the base for RHEL9).

**Comment by Rammurthy5** [comment, marked as answer]
I dont see the error if i use Ubuntu 22.04.

**Comment by dgleba** [comment]
Since this is closed and it did not re-open, I posted here: https://github.com/orgs/pgEdge/discussions/60

-----

I am getting the error..
```
File "/opt/pgedge/pgedge/hub/scripts/spock.py", line 6, in <module>
docker-pgedge-n2-1  |     import util, meta, api, fire, psycopg
docker-pgedge-n2-1  | ModuleNotFoundError: No module named 'psycopg'
```
I get it on all the machines I try. This one is Xubuntu 22.04. 

I am following `https://www.pgedge.com/blog/pgedge-docker-compose-quick-start`

```
python3 -c "$(curl -fsSL https://pgedge-download.s3.amazonaws.com/REPO/install.py)"
cd pgedge/docker
docker compose build
docker compose up
```
```
albe@del-7410:/ap/test/732collection/red74/pgedge74/docker$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
albe@del-7410:/ap/test/732collection/red74/pgedge74/docker$
```

Any hints for me?
